### PR TITLE
Put `ModelBindingHelper` on a diet

### PR DESCRIPTION
**Pull request by dougbu**
- remove extra argument checks
- remove two test-only `ConvertTo()` overloads
 - this relates to #4521

### CODE REVIEW

**Comment by dougbu** [comment]
Note this PR does not "prune" our supported conversions or otherwise fix #4521.

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs [hunk 1]
```diff
@@ -120,41 +85,6 @@ public static class ModelBindingHelper
             params Expression<Func<TModel, object>>[] includeExpressions)
            where TModel : class
         {
-            if (model == null)
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs]: In this particular case, the failure behaviour is slightly different but all `null` parameters are still caught. E.g. if `prefix` and `includeExpressions` were both `null`, would previously get an `ArgumentNullException` complaining about `prefix`. Would now be about `includeExpressions`. Could restore these checks if anyone thinks the order of these checks is important to preserve.

**Comment by dougbu** [comment]
Could someone please self-assign reviewing this small PR?

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs]: Why remove all this code? We do argument validation in the top level method that user code called.

Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs]: It's just noise when the parameters are checked in the longest overload. We aren't consistent about removing duplication but definitely do it.

**Comment by ryanbrandenburg** [comment]
The actual code looks fine to me. I'm of two minds on whether the deduplication is worth not checking the parameters early. They both have pros and cons, so if this is what we do in other parts of the code I say :shipit:.

**Comment by dougbu** [comment]
52e4ca7

**Pull request closed without merging**